include/stl2/view/subrange.hpp

```diff
@@ -109,45 +110,47 @@ STL2_OPEN_NAMESPACE {
 			}
 
 			// Not to spec
-			template <Range R>
-			requires !StoreSize && Iterator<safe_iterator_t<R>> &&
+			template <_NotSameAs<subrange> R>
+			requires Range<R> && Iterator<safe_iterator_t<R>> &&
 				ConvertibleTo<iterator_t<R>, I> && ConvertibleTo<sentinel_t<R>, S>
-			constexpr subrange(R&& r)
+			constexpr subrange(R&& r) requires (!StoreSize)
```

Comment:
Ewww, leading + trailing requires clauses. I assume the intent is to draw attention to the similar and dissimilar requirement sets of the two overloads? I think I buy that as a reason to use both leading and trailing `requires`.

Reply:
For all the members of `subrange` the "sized" constraints are in trailing requires clauses. Makes it a little easier for me to grok if these aren't mixed in with the requirements.

Reply:
I usually try to put non-dependent clauses first, in the hopes that it has (or will have) performance impact. That's probably just wishful thinking on my part.

Reply:
(I've already approved BTW, so don't feel like you *must* address any of these mostly-conversational "issues".)